### PR TITLE
Targeted Imports and Prod Build Fixes

### PR DESCRIPTION
**Pull request by brianghig**
This addresses #17 

- Moves all * imports, where possible, to targeted imports
- Added `bundle-report` NPM script that will create the production distribution
- Updated Travis CI to run production build during execution
- Fixed issue in Admin List Users HTML that caused the production build to fail

### CODE REVIEW

**Comment by brianghig** [comment]
@jrassa @reblace @ymao1 @lu-c the only "big" import we have left is `moment`, which is a dependency from the `AgoDatePipe` and `UtcDatePipe`. If we're okay with the size, then we can leave that alone. However, since `moment` doesn't allow for clean tree-shaking in its current state, we may want to consider moving to something like `date-fns` that has a smaller footprint (as long as it supports the capabilities we still need).

Do many other projects have extensive dependencies on `moment`, or is that falling out of favor?

**Comment by jrassa** [comment]
@brianghig Did you verify that the lodash changes actually resulted in a smaller bundle? I come across [this article](https://medium.com/@armno/til-importing-lodash-into-angular-the-better-way-aacbeaa40473) earlier today that indicates that it will still pull in all of lodash unless we import like this:
```
import { isArray } from 'lodash/isArray';
import { keys } from 'lodash/keys';
import { toString } from 'lodash/toString';
```

**Comment by brianghig** [comment]
@jrassa thanks for that! I had started putting together an analysis of the size reduction, but never finished to post it with this PR. That said...

Good call on your part of looking that up! I fixed the `bundle-report` script to give us granular details of each module, and it's showing that `lodash` is still pulling in the whole thing (as well as plenty of code from `ngx-bootstrap` that we might want to look at.

Looks like there's more to fix with this PR before we call it "done"

screen shot 2018-11-09 at 5 01 16 pm

**Comment by brianghig** [comment]
@jrassa I just updated the lodash imports, and found that they required `import fn from 'lodash/fn';` instead of `import { fn } from 'lodash/fn';`. According to the output from `npm run bundle-report`, this had a nice reduction on the bundle size:

State | Stat Size | Parsed Size | Gzipped
--- | ---- | --- | ---
Before | 527 KB | 527 KB | 93 KB
After | 73 KB | 88 KB | 16 KB
Change | -86.1% | -83.3% | -82.8%

It looks like we have a lot of other opportunities to save even more space from the new reduced baseline of 5.75 MB vendor files.

File: package.json [hunk 1]
```diff
@@ -10,7 +10,8 @@
     "test": "ng test",
     "test:ci": "ng test --watch=false --code-coverage",
     "lint": "ng lint",
-    "e2e": "ng e2e"
+    "e2e": "ng e2e",
+    "bundle-report": "ng build --stats-json && webpack-bundle-analyzer dist/ngx-starter/stats.json"
```
Comment (reblace) [inline package.json]: For consistency, should this script be called "build:bundle-report" or something like that?

Comment (brianghig) [reply package.json]: I like it. I moved it to `build:bundle-report` and added a section into the README that describes each of the available NPM scripts so folks don't have to hunt through the `package.json` file.

**Pull request merged**